docs/transparent-proxy.md

```diff
@@ -0,0 +1,268 @@
+ebtables -t broute -A BROUTING -i enp0s8 -p ipv4 --ip-src $SERVER_IP --ip-proto tcp --ip-source-port 3389 -j redirect --redirect-target DROP
+# recent linux like debian buster
+#ebtables-legacy -t broute -A BROUTING -i enp0s8 -p ipv4 --ip-src $SERVER_IP --ip-proto tcp --ip-source-port 3389 -j redirect --redirect-target DROP
+
+# Disable return path filtering
+echo 0 > /proc/sys/net/ipv4/conf/default/rp_filter
+echo 0 > /proc/sys/net/ipv4/conf/all/rp_filter
+echo 0 > /proc/sys/net/ipv4/conf/enp0s3/rp_filter
+echo 0 > /proc/sys/net/ipv4/conf/enp0s8/rp_filter
+
+# Make interception work without an IP assigned to br0
+# WARNING: unclear if required on not
+echo 1 > /proc/sys/net/ipv4/conf/br0/route_localnet
+echo 1 > /proc/sys/net/ipv4/conf/enp0s3/route_localnet
+
+# Mark socket Packets
+# WARNING: Unclear if required or not but likely good for performance so it will likely stay
+iptables -t mangle -N DIVERT
+iptables -t mangle -A DIVERT -j MARK --set-mark 1
+iptables -t mangle -A DIVERT -j ACCEPT
+iptables -t mangle -I PREROUTING -p tcp -m socket -j DIVERT
+```
+
+In order for L3 traffic to be properly routed between the client and servers, there are two options.
+
+### Option 1: Add an Interface with a valid IP in the networks of interests on the Bridge
+
+This is important to allow ARP resolution to work properly (when the MITM unavoidably needs to know who has the
+server/client IP)
+
+In the example setup, doing something like below is enough to make everything work because when the MITM needs the L2
+address to reply to, it has the means to look it up by itself.
+
+```
+ip addr add 10.13.37.11/24 dev br0
+```
+
+### Option 2: ARP Pinning
+
+To be stealthier, not having any routable L3 address is even better, but this requires to pin ARP cache entries so that
+the MITM is able to resolve any `WHO HAS x, TELL mitm` queries. The good thing about this approach is that the MITM will
+be harder to detect because it will not broadcast ARP at all. The problem with it is that it requires prior knowledge of
+the intended server and of the gateway's MAC and IP addresses, and that any dynamic change to those will invalidate the
+configuration and cause a connectivity outage.
+
+Two ARP entries need to be added:
+
+- The server itself
+- The server's gateway
+
+The command to use to pin an ARP entry is `arp -i br0 -s <ip_addr> <mac_addr>`
+
+In the example above, the two commands might be issued if the gateway for the server is configured as `10.13.37.1`:
+
+```bash
+arp -i br0 -s 10.13.37.1 08:00:27:59:05:f0
+arp -i br0 -s 10.13.37.111 08:00:27:59:05:fe
+```
+
+Lastly, the MITM networking stack needs to know which traffic to route through the bridge, and this can complicate
+things significantly if a management interface is required. The simplest option is to have a route for the
+bridged subnet:
+
+```
+ip route add 10.13.37.0/24 dev br0
+```
```

Comment:
I'm not convinced this is the correct explanation since a bridge just bridges the stuff through no matter what. I think it matters when we are putting packets (spoofed using `IP_TRANSPARENT`) in the kernel then the kernel knows where (at interface-level) to send them or what rules they need to go through (like mangle with pre-routing iface set to br0).